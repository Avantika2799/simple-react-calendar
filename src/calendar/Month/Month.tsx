import React, { FC, Fragment, SyntheticEvent } from 'react'

import {
  IDateRange,
  IMonthHeaderButtonRenderProps,
  IMonthHeaderDayRenderProps,
  IMonthHeaderRenderProps,
  IMonthRenderProps,
  IWeekDayRenderProps,
  IWeekRenderProps,
} from '../../@types'
import { IMonthWeeks } from '../helper'
import MonthHeader from '../MonthHeader'
import Week from '../Week'

export type Props = {
  activeMonth: Date
  blockClassName: string
  customRender?: IMonthRenderProps
  getDayEachDay: ({ start, end }: IDateRange) => Date[]
  getDayFormatted: (date: Date) => string
  getDayIsCurrentMonth: (date: Date) => boolean
  getDayIsDisabled: (date: Date) => boolean
  getDayIsHighlighted: (date: Date) => boolean
  getDayIsNextMonth: (date: Date) => boolean
  getDayISOFormatter: (date: Date) => string
  getDayIsSelectable: (date: Date) => boolean
  getDayIsSelected: (date: Date) => boolean
  getDayIsSelectedEnd: (date: Date) => boolean
  getDayIsSelectedStart: (date: Date) => boolean
  getDayIsToday: (date: Date) => boolean
  getDayIsWeekend: (date: Date) => boolean
  getDayIndexIsWeekend: (index: 0 | 1 | 2 | 3 | 4 | 5 | 6) => boolean
  getMonthFormattedName: (date: Date) => string
  getWeekDays: () => string[]
  getMonthWeeks: (activeMonth: Date) => IMonthWeeks[]
  handleOnDayClick: (event: SyntheticEvent<HTMLButtonElement>) => void
  handleOnDayMouseEnter: (event: SyntheticEvent<HTMLButtonElement>) => void
  handleOnDisabledDayClick: (event: SyntheticEvent<HTMLButtonElement>) => void
  handleOnDisabledDayMouseEnter: (event: SyntheticEvent<HTMLButtonElement>) => void
  handleOnSwitchMonth: (event: SyntheticEvent<HTMLElement>) => void
  isMonthNextEnabled: boolean
  isMonthPrevEnabled: boolean
  renderMonthHeader?: IMonthHeaderRenderProps
  renderMonthHeaderButton?: IMonthHeaderButtonRenderProps
  renderMonthHeaderDay?: IMonthHeaderDayRenderProps
  renderWeek?: IWeekRenderProps
  renderWeekDay?: IWeekDayRenderProps
  showDaysOfWeek: boolean
  showMonthName: boolean
  showMonthNext: boolean
  showMonthPrev: boolean
}

const Month: FC<Props> = (props) => {
  const {
    activeMonth,
    blockClassName,
    customRender,
    getDayEachDay,
    getDayFormatted,
    getDayIsCurrentMonth,
    getDayIsDisabled,
    getDayIsHighlighted,
    getDayIsNextMonth,
    getDayISOFormatter,
    getDayIsSelectable,
    getDayIsSelected,
    getDayIsSelectedEnd,
    getDayIsSelectedStart,
    getDayIsToday,
    getDayIsWeekend,
    getDayIndexIsWeekend,
    getMonthFormattedName,
    getWeekDays,
    getMonthWeeks,
    handleOnDayClick,
    handleOnDayMouseEnter,
    handleOnDisabledDayClick,
    handleOnDisabledDayMouseEnter,
    handleOnSwitchMonth,
    isMonthNextEnabled,
    isMonthPrevEnabled,
    renderMonthHeader,
    renderMonthHeaderButton,
    renderMonthHeaderDay,
    renderWeek,
    renderWeekDay,
    showDaysOfWeek,
    showMonthName,
    showMonthNext,
    showMonthPrev,
  } = props

  const children = (
    <Fragment>
      <MonthHeader
        activeMonth={activeMonth}
        blockClassName={blockClassName}
        customRender={renderMonthHeader}
        getDayIndexIsWeekend={getDayIndexIsWeekend}
        getMonthFormattedName={getMonthFormattedName}
        getWeekDays={getWeekDays}
        handleOnSwitchMonth={handleOnSwitchMonth}
        isMonthNextEnabled={isMonthNextEnabled}
        isMonthPrevEnabled={isMonthPrevEnabled}
        renderMonthHeaderButton={renderMonthHeaderButton}
        renderMonthHeaderDay={renderMonthHeaderDay}
        showDaysOfWeek={showDaysOfWeek}
        showMonthName={showMonthName}
        showMonthNext={showMonthNext}
        showMonthPrev={showMonthPrev}
      />
      {getMonthWeeks(activeMonth).map(({ weekStart, weekNumber, weekEnd }: IMonthWeeks) => (
        <Week
          blockClassName={blockClassName}
          customRender={renderWeek}
          getDayEachDay={getDayEachDay}
          getDayFormatted={getDayFormatted}
          getDayIsCurrentMonth={getDayIsCurrentMonth}
          getDayIsDisabled={getDayIsDisabled}
          getDayIsHighlighted={getDayIsHighlighted}
          getDayIsNextMonth={getDayIsNextMonth}
          getDayISOFormatter={getDayISOFormatter}
          getDayIsSelectable={getDayIsSelectable}
          getDayIsSelected={getDayIsSelected}
          getDayIsSelectedEnd={getDayIsSelectedEnd}
          getDayIsSelectedStart={getDayIsSelectedStart}
          getDayIsToday={getDayIsToday}
          getDayIsWeekend={getDayIsWeekend}
          handleOnDayClick={handleOnDayClick}
          handleOnDayMouseEnter={handleOnDayMouseEnter}
          handleOnDisabledDayClick={handleOnDisabledDayClick}
          handleOnDisabledDayMouseEnter={handleOnDisabledDayMouseEnter}
          key={weekNumber}
          renderWeekDay={renderWeekDay}
          weekEnd={weekEnd}
          weekNumber={weekNumber}
          weekStart={weekStart}
        />
      ))}
    </Fragment>
  )

  if (customRender) {
    return customRender(props)
  }

  return <div className={`${blockClassName}-month`}>{children}</div>
}

Month.displayName = 'Month'

export default Month
