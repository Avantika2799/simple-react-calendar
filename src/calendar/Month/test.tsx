import { cleanup, fireEvent, render, RenderResult } from '@testing-library/react'
import React from 'react'

import { IDateRange, ISelectionRange } from '../../@types'
import {
  getDayEachDay,
  getDayFormatted,
  getDayIndexIsWeekend,
  getDayIsCurrentMonth,
  getDayIsDisabled,
  getDayIsHighlighted,
  getDayIsNextMonth,
  getDayISOFormatter,
  getDayIsSame,
  getDayIsSelectable,
  getDayIsSelected,
  getDayIsSelectedEnd,
  getDayIsSelectedStart,
  getDayIsWeekend,
  getMonthFormattedName,
  getMonthWeeks,
  getWeekDays,
} from '../helper'
import Month, { Props } from './Month'

afterEach(cleanup)

describe('Month', () => {
  let api: RenderResult,
    activeMonth: Date,
    disabledIntervals: IDateRange[],
    maxDate: Date,
    minDate: Date,
    mockedCustomRender: any,
    mode: 'range' | 'single',
    props: Props,
    selected: ISelectionRange,
    today: Date,
    weekStartsOn: 0 | 1 | 2 | 3 | 4 | 5 | 6

  beforeEach(() => {
    activeMonth = new Date(2019, 10, 1)
    disabledIntervals = []
    maxDate = new Date(2019, 10, 26)
    minDate = new Date(2019, 10, 20)
    mode = 'range'
    selected = { inProgress: false, start: new Date(2019, 10, 28), end: new Date(2019, 10, 29) }
    today = new Date(2019, 10, 27)
    weekStartsOn = 1

    props = {
      activeMonth,
      blockClassName: 'test-class-name',
      getDayEachDay,
      getDayFormatted,
      getDayIsCurrentMonth: (date) => getDayIsCurrentMonth({ activeMonth, date }),
      getDayIsDisabled: (date) => getDayIsDisabled({ date, disabledIntervals }),
      getDayIsHighlighted: (date) => getDayIsHighlighted({ date, highlightedIntervals: [] }),
      getDayIsNextMonth: (date) => getDayIsNextMonth({ activeMonth, date }),
      getDayISOFormatter,
      getDayIsSelectable: (date) => getDayIsSelectable({ date, minDate, maxDate, disabledIntervals }),
      getDayIsSelected: (date) => getDayIsSelected({ date, mode, selected }),
      getDayIsSelectedEnd: (date) => getDayIsSelectedStart({ date, mode, selected }),
      getDayIsSelectedStart: (date) => getDayIsSelectedEnd({ date, mode, selected }),
      getDayIsToday: (date) => getDayIsSame({ date, dateToCompare: today }),
      getDayIsWeekend: (date) => getDayIsWeekend({ date }),
      getDayIndexIsWeekend: (index) =>
        getDayIndexIsWeekend({ dayIndex: index as 0 | 1 | 2 | 3 | 4 | 5 | 6, weekStartsOn }),
      getMonthWeeks: () => getMonthWeeks({ activeMonth, weekStartsOn }),
      getMonthFormattedName,
      getWeekDays: () => getWeekDays({ daysOfWeek: ['Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat', 'Sun'], dayIndex: 1 }),
      handleOnDayClick: jest.fn(),
      handleOnDayMouseEnter: jest.fn(),
      handleOnDisabledDayClick: jest.fn(),
      handleOnDisabledDayMouseEnter: jest.fn(),
      handleOnSwitchMonth: jest.fn(),
      isMonthNextEnabled: true,
      isMonthPrevEnabled: false,
      showDaysOfWeek: true,
      showMonthName: true,
      showMonthNext: true,
      showMonthPrev: true,
    }
  })

  describe('action handlers', () => {
    beforeEach(() => {
      api = render(<Month {...props} />)
    })

    test('invoke `#isMonthNextEnabled`', () => {
      const { getByTitle } = api

      fireEvent.click(getByTitle(/Next Month/i))

      expect(props.handleOnSwitchMonth).toHaveBeenCalledTimes(1)
    })
  })

  describe('when render prop has not been passed', () => {
    beforeEach(() => {
      api = render(<Month {...props} />)
    })

    test('default render', () => {
      const { container } = api

      expect(container).toMatchSnapshot()
    })

    describe('when `showMonthPrev`, `showMonthName`, `showMonthNext` `showDaysOfWeek` props are `falsy`', () => {
      beforeEach(() => {
        props = {
          ...props,
          showMonthPrev: false,
          showMonthName: false,
          showMonthNext: false,
          showDaysOfWeek: false,
        }
        api = render(<Month {...props} />)
      })

      test('default render', () => {
        const { container } = api

        expect(container).toMatchSnapshot()
      })
    })
  })

  describe('when render prop has been passed', () => {
    beforeEach(() => {
      mockedCustomRender = jest.fn(() => null)

      props = {
        ...props,
        customRender: mockedCustomRender,
      }
      api = render(<Month {...props} />)
    })

    test('render prop has been called properly', () => {
      expect(props.customRender).toHaveBeenCalledTimes(1)
      expect(props.customRender).toHaveBeenCalledWith(props)
    })
  })
})
