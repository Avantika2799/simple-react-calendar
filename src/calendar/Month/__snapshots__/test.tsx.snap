// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Month when render prop has not been passed default render 1`] = `
<div>
  <div
    class="test-class-name-month"
  >
    <header
      class="test-class-name-month_header"
    >
      <div
        class="test-class-name-month_header_wrapper"
      >
        <button
          class="test-class-name-header_button is-prev is-disabled"
          data-calendar-value="-1"
          disabled=""
          title="Previous Month"
          type="button"
        >
          &gt;
        </button>
        <h3
          class="test-class-name-month_header_title"
        >
          November 2019
        </h3>
        <button
          class="test-class-name-header_button is-next"
          data-calendar-value="1"
          title="Next Month"
          type="button"
        >
          &lt;
        </button>
      </div>
      <div
        class="test-class-name-month_header_days"
      >
        <div
          class="test-class-name-days_of_week_day is-weekend"
        >
          Mon
        </div>
        <div
          class="test-class-name-days_of_week_day"
        >
          Tue
        </div>
        <div
          class="test-class-name-days_of_week_day"
        >
          Wed
        </div>
        <div
          class="test-class-name-days_of_week_day"
        >
          Thu
        </div>
        <div
          class="test-class-name-days_of_week_day"
        >
          Fri
        </div>
        <div
          class="test-class-name-days_of_week_day"
        >
          Sat
        </div>
        <div
          class="test-class-name-days_of_week_day is-weekend"
        >
          Sun
        </div>
      </div>
    </header>
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        44
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-not_selectable is-prev_month is-working_day"
            type="button"
            value="2019-10-28"
          >
            28
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-not_selectable is-prev_month is-working_day"
            type="button"
            value="2019-10-29"
          >
            29
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-not_selectable is-prev_month is-working_day"
            type="button"
            value="2019-10-30"
          >
            30
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-not_selectable is-prev_month is-working_day"
            type="button"
            value="2019-10-31"
          >
            31
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-01"
          >
            1
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-02"
          >
            2
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-03"
          >
            3
          </button>
        </div>
      </div>
    </div>
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        45
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-04"
          >
            4
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-05"
          >
            5
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-06"
          >
            6
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-07"
          >
            7
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-08"
          >
            8
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-09"
          >
            9
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-10"
          >
            10
          </button>
        </div>
      </div>
    </div>
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        46
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-11"
          >
            11
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-12"
          >
            12
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-13"
          >
            13
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-14"
          >
            14
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-15"
          >
            15
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-16"
          >
            16
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-17"
          >
            17
          </button>
        </div>
      </div>
    </div>
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        47
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-18"
          >
            18
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-19"
          >
            19
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-20"
          >
            20
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-21"
          >
            21
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-22"
          >
            22
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-weekend"
            type="button"
            value="2019-11-23"
          >
            23
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-weekend"
            type="button"
            value="2019-11-24"
          >
            24
          </button>
        </div>
      </div>
    </div>
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        48
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-25"
          >
            25
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-26"
          >
            26
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-today is-working_day"
            type="button"
            value="2019-11-27"
          >
            27
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-end_selection is-not_selectable is-selected is-working_day"
            type="button"
            value="2019-11-28"
          >
            28
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-selected is-start_selection is-working_day"
            type="button"
            value="2019-11-29"
          >
            29
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-30"
          >
            30
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-next_month is-not_selectable is-weekend"
            type="button"
            value="2019-12-01"
          >
            1
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Month when render prop has not been passed when \`showMonthPrev\`, \`showMonthName\`, \`showMonthNext\` \`showDaysOfWeek\` props are \`falsy\` default render 1`] = `
<div>
  <div
    class="test-class-name-month"
  >
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        44
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-not_selectable is-prev_month is-working_day"
            type="button"
            value="2019-10-28"
          >
            28
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-not_selectable is-prev_month is-working_day"
            type="button"
            value="2019-10-29"
          >
            29
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-not_selectable is-prev_month is-working_day"
            type="button"
            value="2019-10-30"
          >
            30
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-not_selectable is-prev_month is-working_day"
            type="button"
            value="2019-10-31"
          >
            31
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-01"
          >
            1
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-02"
          >
            2
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-03"
          >
            3
          </button>
        </div>
      </div>
    </div>
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        45
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-04"
          >
            4
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-05"
          >
            5
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-06"
          >
            6
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-07"
          >
            7
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-08"
          >
            8
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-09"
          >
            9
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-10"
          >
            10
          </button>
        </div>
      </div>
    </div>
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        46
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-11"
          >
            11
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-12"
          >
            12
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-13"
          >
            13
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-14"
          >
            14
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-15"
          >
            15
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-16"
          >
            16
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-17"
          >
            17
          </button>
        </div>
      </div>
    </div>
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        47
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-18"
          >
            18
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-working_day"
            type="button"
            value="2019-11-19"
          >
            19
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-20"
          >
            20
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-21"
          >
            21
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-22"
          >
            22
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-weekend"
            type="button"
            value="2019-11-23"
          >
            23
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-weekend"
            type="button"
            value="2019-11-24"
          >
            24
          </button>
        </div>
      </div>
    </div>
    <div
      class="test-class-name-week"
    >
      <div
        class="test-class-name-week_number"
      >
        48
      </div>
      <div
        class="test-class-name-week_days"
      >
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-25"
          >
            25
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-selectable is-working_day"
            type="button"
            value="2019-11-26"
          >
            26
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-today is-working_day"
            type="button"
            value="2019-11-27"
          >
            27
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-end_selection is-not_selectable is-selected is-working_day"
            type="button"
            value="2019-11-28"
          >
            28
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-selected is-start_selection is-working_day"
            type="button"
            value="2019-11-29"
          >
            29
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-current_month is-not_selectable is-weekend"
            type="button"
            value="2019-11-30"
          >
            30
          </button>
        </div>
        <div
          class="test-class-name-day_wrapper"
        >
          <button
            class="test-class-name-day is-next_month is-not_selectable is-weekend"
            type="button"
            value="2019-12-01"
          >
            1
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
